torchaudio/models/decoder/_ctc_decoder.py

```diff
@@ -270,12 +282,22 @@ def ctc_decoder(
         >>> )
         >>> results = decoder(emissions) # List of shape (B, nbest) of Hypotheses
     """
+    if lm_dict is not None and type(lm_dict) is not str:
+        raise ValueError("lm_dict must be None or str type.")
+
     tokens_dict = _Dictionary(tokens)
+    if lm_dict:
```

Comment:
it's a little expensive (esp in python so probably not worth doing), but a restriction in the decoder for obvious reasons is that LM dictionaries must only have words which are in lexicons [if both lexicons and NNLM dicts are provided]. Flashlight will throw an exception internally if this isn't the case iirc, but perhaps worth noting in docs.